Content.Server/SurveillanceCamera/Systems/SurveillanceCameraRouterSystem.cs

```diff
@@ -0,0 +1,269 @@
+        SubscribeLocalEvent<SurveillanceCameraRouterComponent, PowerChangedEvent>(OnPowerChanged);
+    }
+
+    private void OnInitialize(EntityUid uid, SurveillanceCameraRouterComponent router, ComponentInit args)
+    {
+        if (router.SubnetFrequencyId == null ||
+            !_prototypeManager.TryIndex(router.SubnetFrequencyId, out DeviceFrequencyPrototype? subnetFrequency))
+        {
+            return;
+        }
+
+        router.SubnetFrequency = subnetFrequency.Frequency;
+    }
+
+    private void OnPacketReceive(EntityUid uid, SurveillanceCameraRouterComponent router, DeviceNetworkPacketEvent args)
+    {
+        if (!router.Active
+            || string.IsNullOrEmpty(args.SenderAddress)
+            || !args.Data.TryGetValue(DeviceNetworkConstants.Command, out string? command))
+        {
+            return;
+        }
+
+        switch (command)
+        {
+            case SurveillanceCameraSystem.CameraConnectMessage:
+                if (!args.Data.TryGetValue(SurveillanceCameraSystem.CameraAddressData, out string? address))
+                {
+                    return;
+                }
+
+                ConnectCamera(uid, args.SenderAddress, address, router);
+                break;
+            case SurveillanceCameraSystem.CameraHeartbeatMessage:
+                if (!args.Data.TryGetValue(SurveillanceCameraSystem.CameraAddressData, out string? camera))
+                {
+                    return;
+                }
+
+                SendHeartbeat(uid, args.SenderAddress, camera, router);
+                break;
+            case SurveillanceCameraSystem.CameraSubnetConnectMessage:
+                AddMonitorToRoute(uid, args.SenderAddress, router);
+                PingSubnet(uid, router);
+                break;
+            case SurveillanceCameraSystem.CameraSubnetDisconnectMessage:
+                RemoveMonitorFromRoute(uid, args.SenderAddress, router);
+                break;
+            case SurveillanceCameraSystem.CameraPingSubnetMessage:
+                PingSubnet(uid, router);
+                break;
+            case SurveillanceCameraSystem.CameraPingMessage:
+                SubnetPingResponse(uid, args.SenderAddress, router);
+                break;
+            case SurveillanceCameraSystem.CameraDataMessage:
+                SendCameraInfo(uid, args.Data, router);
+                break;
+        }
+    }
+
+    private void OnPowerChanged(EntityUid uid, SurveillanceCameraRouterComponent component, PowerChangedEvent args)
+    {
+        component.Active = args.Powered;
+    }
+
+    private void AddVerbs(EntityUid uid, SurveillanceCameraRouterComponent component, GetVerbsEvent<Verb> verbs)
+    {
+        if (!TryComp(verbs.User, out ActorComponent? actor))
+        {
+            return;
+        }
+
+        if (HasComp<GhostComponent>(verbs.User) && !_adminManager.IsAdmin(actor.PlayerSession))
+        {
+            return;
+        }
```

Comment:
Do any of the actionblocker checks already check for this, e.g. CanInteract?

Reply:
unsure: admin ghosts have the ability to interact with things, right? if so i can remove this check

Reply:
Yeah they do; they have hands.